mobilitypath_visualizer/src/mobilitypath_visualizer.cpp

```diff
@@ -32,96 +33,143 @@ namespace mobilitypath_visualizer {
         }
     }
 
-    MobilityPathVisualizer::MobilityPathVisualizer() = default;
-
-    void MobilityPathVisualizer::run()
+    MobilityPathVisualizer::MobilityPathVisualizer(const rclcpp::NodeOptions &options) 
+            : carma_ros2_utils::CarmaLifecycleNode(options)
     {
-        initialize();
-        ros::CARMANodeHandle::setSpinRate(spin_rate_);
-        ros::CARMANodeHandle::setSpinCallback(std::bind(&MobilityPathVisualizer::spinCallback, this));
-        ros::CARMANodeHandle::spin();
+        // Create initial config
+        config_ = Config();
+
+        // Declare parameters
+        config_.timer_cb_rate = declare_parameter<int>("timer_cb_rate", config_.timer_cb_rate);
+        config_.x = declare_parameter<double>("x", config_.x);
+        config_.y = declare_parameter<double>("y", config_.y);
+        config_.z = declare_parameter<double>("z", config_.z);
+        config_.t = declare_parameter<double>("t", config_.t);
     }
 
-    void MobilityPathVisualizer::initialize()
+
+    carma_ros2_utils::CallbackReturn MobilityPathVisualizer::handle_on_configure(const rclcpp_lifecycle::State &)
     {
-        // init CARMANodeHandle
-        nh_.reset(new ros::CARMANodeHandle());
-        pnh_.reset(new ros::CARMANodeHandle("~"));
-        pnh_->param<double>("spin_rate", spin_rate_, 10.0);
-        pnh_->param<double>("x", x_, 0.5);
-        pnh_->param<double>("y", y_, 0.5);
-        pnh_->param<double>("z", z_, 1.0);
-        pnh_->param<double>("t", t_, 3.0);
+        //Reset Config
+        config_ = Config();
+        
+        //Load parameters
+        get_parameter<int>("timer_cb_rate", config_.timer_cb_rate);
+        get_parameter<double>("x", config_.x);
+        get_parameter<double>("y", config_.y);
+        get_parameter<double>("z", config_.z);
+        get_parameter<double>("t", config_.t);
 
-        nh_->getParam("/vehicle_id", host_id_);
+        get_parameter("/vehicle_id", host_id_);
```

Comment:
Since the specific parameter .yaml file is used when launching this node from environment.launch.py now (rather than this parameter being a global parameter), I believe we can remove the forward-slash here and the parameter can simply be read in as `"vehicle_id"`